lib/spack/spack/cmd/mirror.py

```diff
@@ -45,6 +45,10 @@ def setup_parser(subparser):
              " (this requires significant time and space)")
     create_parser.add_argument(
         '-f', '--file', help="file with specs of packages to put in mirror")
+    create_parser.add_argument(
```

Comment:
I think this should also have a command-line option to list the excluded specs directly

Reply:
OK, I'm assuming this should look like

```
spack mirror create ... --exclude-specs=foo,bar,baz
```

? Another possibility is to provide an `--exclude` flag and the spec args are interpreted to be excludes. In that case, `--exclude` would only be useful as an argument when `--all` is specified.

Reply:
I think `spack.cmd.parse_specs` already can parse multiple specs, so there's no need to include commas, but you will need quotes so that argparse can know when the arguments to `--exclude-specs` is done and it can parse the rest of the API. Something like

`spack mirror create ... --exclude-specs="foo %gcc bar@1.2.0 cflags=-g baz" ...`

Reply:
Great - thanks for the suggestion: I updated the PR with that option (and tests for it).